src/components/Calculator.js

```diff
@@ -1,31 +1,50 @@
+/* eslint-disable react/no-access-state-in-setstate */
+/* eslint-disable react/no-unused-state */
+/* eslint-disable react/destructuring-assignment */
 /* eslint-disable react/button-has-type */
 /* eslint-disable react/prefer-stateless-function */
 import React from 'react';
+import calculate from '../logic/calculate';
 
 export default class MyCalculator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { total: 0, next: 0, operation: '' };
+    this.calchandler = this.calchandler.bind(this);
+  }
+
+  calchandler(e) {
+    if (e.target.innerText === 'AC') {
+      const { next, operation } = calculate(this.state, e.target.innerText);
+      this.setState({ total: 0, next, operation });
+    } else {
+      this.setState(calculate(this.state, e.target.innerText));
+    }
+  }
+
   render() {
     return (
       <div className="grid-continer">
-        <input className="resultscreen" placeholder="0" />
-        <button>AC</button>
-        <button>+/-</button>
-        <button>%</button>
-        <button className="buttoperation">/</button>
-        <button>7</button>
-        <button>8</button>
-        <button>9</button>
-        <button className="buttoperation">x</button>
-        <button>4</button>
-        <button>5</button>
-        <button>6</button>
-        <button className="buttoperation">-</button>
-        <button>1</button>
-        <button>2</button>
-        <button>3</button>
-        <button className="buttoperation">+</button>
-        <button className="zeroitem">0</button>
-        <button>.</button>
-        <button className="buttoperation">=</button>
+        <input className="resultscreen" readOnly placeholder={this.state.total} />
```

Comment:
- [ ] Kindly double-check on the rendered values to the screen of the calculator, it seems that the numbers aren't rendering to the screen while operating. Only the results are! Please let's work on that 🙏 😄